cmake/OpenCVFindLAPACK.cmake

```
@@ -1,3 +1,26 @@
+if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+  set(_apple_device_min_target_os_version "13.3")
+elseif(CMAKE_SYSTEM_NAME STREQUAL "iOS")
+  set(_apple_device_min_target_os_version "16.4")
+elseif(CMAKE_SYSTEM_NAME STREQUAL "watchOS")
+  set(_apple_device_min_target_os_version "9.4")
+elseif(CMAKE_SYSTEM_NAME STREQUAL "tvOS")
+  set(_apple_device_min_target_os_version "16.4")
+elseif(CMAKE_SYSTEM_NAME STREQUAL "visionOS")
+  set(_apple_device_min_target_os_version "1.0")
+endif()
+
+if(DEFINED _apple_device_min_target_os_version AND
+   ("${CMAKE_OSX_DEPLOYMENT_TARGET}" VERSION_GREATER "${_apple_device_min_target_os_version}" OR
+    "${CMAKE_OSX_DEPLOYMENT_TARGET}" VERSION_EQUAL "${_apple_device_min_target_os_version}"))
```

Condition can be simplified

```suggestion
if(_apple_device_min_target_os_version
  AND (${CMAKE_OSX_DEPLOYMENT_TARGET} VERSION_GREATER ${_apple_device_min_target_os_version}))
```

---

[`VERSION_GREATER`](https://cmake.org/cmake/help/latest/command/if.html#version-greater-equal) requires CMake >= 3.7 but OpenCV 4.x has [default minimum requirement of CMake >= 3.5.1](https://github.com/opencv/opencv/blob/79af357cf44214abde34c2f1ac79916849d179ce/cmake/OpenCVMinDepVersions.cmake#L2).